### PR TITLE
mysqlctld: setup a different default for onterm_timeout

### LINKED ISSUE #15577: Bug Report: Default on term timeout for mysqctld is broken
### Overview of the Issue

The default timeout for `OnTerm` hooks is only 10 seconds. This is way too low for `mysqlctld`, since it might also need to shut down MySQL itself. There's already a shutdown configuration for that which defaults to 5 minutes, but in case of a `term` signal, we only ever wait 10 seconds. 

### Reproduction Steps

Send a `term` signal to `mysqlctld` when it manages a large MySQL instance and see the process shuts down after 10 seconds. 

### Binary Version

```sh
-
```

### Operating System and Environment details

```sh
-
```

### Log Fragments

_No response_

### PR DESCRIPTION
**Pull request by dbussink**
Until now we've had the same timeouts for shutdown etc. across the different binaries. This doesn't provide the necessary flexibility though, since for example `mysqlctld` will need longer timeouts.

This is because it needs to shut down MySQL and we need it to honor the MySQL shutdown timeout as well. This means this can take significantly longer to ensure the shutdown is clean.

It's necessary to do this, to ensure things like being able to upgrade MySQL since that depends on clean shutdowns.

## Related Issue(s)

Fixes #15577

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/flags/endtoend/mysqlctld.txt
Comment (deepthi) [inline go/flags/endtoend/mysqlctld.txt:106]: I assume this value is set to be 10 seconds greater than the shutdown timeout defined at
https://github.com/vitessio/vitess/blob/37e5c4ac9b1b93e1e363cb8470ecedc2b80c195d/go/vt/vttablet/tabletmanager/tm_init.go#L95

It may be better to define a global constant for 5 minutes and use it in both places to prevent drift in future.

**Review by deepthi — COMMENTED** [review]
I am onboard with this change. However it is technically a backwards incompatible change, so we need to document it as such.
Let's create an issue and write a release note. I believe website flag docs will also need to be regenerated unless that is now automatically triggered by merges on the main repo.

File: go/flags/endtoend/mysqlctld.txt
Comment (dbussink) [reply go/flags/endtoend/mysqlctld.txt:106]: Not sure what you mean here. The additional 10 seconds was deliberate for some extra buffer time, so not the same for both. 

What would you suggest?

Comment (deepthi) [reply go/flags/endtoend/mysqlctld.txt:106]: something like:
```
var SharedTimeout = 5*time.Minute
...
mysqlShutdownTimeout = SharedTimeout
...
mysqlctldOnTermTimeout = SharedTimeout + 10*time.Second
```
It looks really ugly written out like this, but it establishes a correlation in code between these two timeouts instead of that being in our heads.
Of course now that I think about it, it won't prevent someone from changing mysqlShutdownTimeout to some other value anyway. So I'll leave it up to you whether we should really correlate these two timeouts in code in an explicit way.
A simpler way is to make `mysqlShutdownTimeout` exported, i.e. change it to `MysqlShutdownTimeout` and use it directly as the base for mysqlctld's on term timeout.
🤷

Comment (dbussink) [reply go/flags/endtoend/mysqlctld.txt:106]: I think the current version makes the dependency more clear that the timeout is how long we need for MySQL shutdown and a buffer? That’s imho less clear in that example and to me looks more accidental instead of deliberate.

Comment (dbussink) [reply go/flags/endtoend/mysqlctld.txt:106]: @deepthi Ah wait, I think I get the confusion. It's not really based on what `tm_init.go` has, but what is already in `mysqlctld`. But those could be shared, so it's less about this new change I think.

Comment (dbussink) [reply go/flags/endtoend/mysqlctld.txt:106]: @deepthi Did a little refactor in https://github.com/vitessio/vitess/pull/15575/commits/f1a508fc25ea76e219e7563fdd36aa25dc159d4e

File: go/cmd/vtbackup/cli/vtbackup.go
Comment (deepthi) [inline go/cmd/vtbackup/cli/vtbackup.go:90]: Insane that we were defining this 5 minutes in multiple places.

**Review by deepthi — COMMENTED** [review]
LGTM. Let's get the issue created and release note added, and I can approve after I review the release note.
For the record, I think it is indeed a good idea to start moving globals at the servenv level into structs that can be overridden by each binary, so thank you for starting us down this path.

**Comment by deepthi** [comment]
> I believe website flag docs will also need to be regenerated unless that is now automatically triggered by merges on the main repo.

They are generated by vitess-bot, so I've remove the label for website docs.

**Comment by dbussink** [comment]
@deepthi Added a release note as well.

**Review by frouioui — APPROVED** [review]

**Review by deepthi — APPROVED** [review]

**Pull request merged**
